openvdb_points/tools/PointDataGrid.h

```diff
@@ -360,6 +360,38 @@ class PointDataLeafNode : public tree::LeafNode<T, Log2Dim> {
         mAttributeSet->reorderAttributes(replacement);
     }
 
+    /// @brief Swap the underlying attribute set with the given @a attributeSet.
+    /// This leaf will assume ownership of the given attribute set. The new
+    /// attributes must have an equal descriptor. Additionally, if the new
+    /// attributes have different topology, then the voxel offset values must
+    /// be updated to reflect this new topology after a call to this method.
+    void swap(AttributeSet* attributeSet)
+    {
+        if (mAttributeSet->descriptor() != attributeSet->descriptor()) {
+            OPENVDB_THROW(ValueError, "Attribute set descriptors are not equal.");
+        }
+
+        mAttributeSet.reset(attributeSet);
+    }
+
+    /// @brief Sets all of the voxel offset values on this leaf, from the given vector
+    /// of @a offsets. If @a updateActiveVoxelMask is true, then the active voxel mask will
+    /// also be updated such that voxels with points are marked as active, and empty voxels
+    /// are marked as inactive.
+    void setOffsets(const std::vector<ValueType>& offsets, bool updateActiveVoxelMask = true)
```

Comment:
Maybe we just stick with the same name as the function, updateValueMask. (slightly annoying because it clashes with the function name though)